test/common/init/target_impl_test.cc

```diff
@@ -60,6 +62,110 @@ TEST(InitTargetImplTest, ReadyWhenWatcherUnavailable) {
   EXPECT_FALSE(target.ready());
 }
 
+TEST(SharedTargetImplTest, OnSyncManagers) {
+  ManagerImpl m1("m1");
+  ManagerImpl m2("m2");
+  int counter = 0;
+  SharedTargetImpl t("test", [&counter]() mutable { counter++; });
```

Comment:
This pattern is copy-pasted across all these tests. Can you extract it into a helper class similar to `ExpectableTargetImpl`?

Reply:
will do